Doc/library/shutil.rst

```diff
@@ -370,6 +395,26 @@ Directory and files operations
    operation. For :func:`copytree`, the exception argument is a list of 3-tuples
    (*srcname*, *dstname*, *exception*).
 
+.. _shutil-platform-dependent-efficient-copy-operations:
+
+Platform-dependent efficient copy operations
+~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+
+Starting from Python 3.8 :func:`copyfile` uses platform-specific "zero-copy"
+syscalls such as :func:`os.sendfile` in order to copy the file more efficiently
+(see `bpo-33671 <https://bugs.python.org/issue33671>`_).
+"zero-copy" means that the copying operation occurs within the kernel, avoiding
+the use of userspace buffers as in "``outfd.write(infd.read())``".
+Such platforms are OSX, Windows and POSIX systems where :func:`os.sendfile`
+accepts 2 regular fds (namely Linux and Solaris).
+If the zero-copy operation fails and no data was written in the destination
+file then :func:`copyfile` will silently fallback on using less efficient
+:func:`copyfileobj` function internally.
+All functions relying on :func:`copyfile` will benefit from the same speedup.
+These are :func:`shutil.copy`, :func:`shutil.copy2`, :func:`shutil.copytree`
+and :func:`shutil.move`.
+
```

Comment:
Given @eryksun's comments about how the relevant Windows API works, it may be better to just name the platform specific APIs here, without going into details on exactly how those APIs achieve their performance gains relative to a Python level copy operation.